bin/prover-client/src/task_tracker.rs

```diff
@@ -170,6 +181,25 @@ impl TaskTracker {
             })
             .collect()
     }
+
+    /// Generates a report of task statuses and their counts across all tasks.
+    pub fn generate_report(&self) -> HashMap<String, usize> {
+        let mut report: HashMap<String, usize> = HashMap::new();
+
+        for status in self.tasks.values() {
+            let status_str = match status {
+                ProvingTaskStatus::WaitingForDependencies(_) => "WaitingForDependencies",
+                ProvingTaskStatus::Pending => "Pending",
+                ProvingTaskStatus::ProvingInProgress => "ProvingInProgress",
+                ProvingTaskStatus::Completed => "Completed",
+                ProvingTaskStatus::Failed => "Failed",
```

Comment:
Could we move the deps out of the status field so that we can just `Debug`/`Display` it?  If we kept the deps list as part of some task control header then that would let us visualize the trees a bit better retrospectively.